tests/integration/tests/test_util/snap.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import json
+import logging
+import re
+import urllib.error
+import urllib.request
+from typing import List
+
+LOG = logging.getLogger(__name__)
+
+SNAP_NAME = "k8s"
+
+# For Snap Store API request
+SNAPSTORE_API = "https://api.snapcraft.io/v2/snaps/info/"
+SNAPSTORE_HEADERS = {
+    "Snap-Device-Series": "16",
+    "User-Agent": "Mozilla/5.0",
+}
+RISK_LEVELS = ["stable", "candidate", "beta", "edge"]
+
+
+def get_snap_info(snap_name=SNAP_NAME):
+    """Get the snap info from the Snap Store API."""
+    req = urllib.request.Request(SNAPSTORE_API + snap_name, headers=SNAPSTORE_HEADERS)
+    try:
+        with urllib.request.urlopen(req) as response:  # nosec
+            return json.loads(response.read().decode())
+    except urllib.error.HTTPError as e:
+        LOG.exception("HTTPError ({%s}): {%s} {%s}", req.full_url, e.code, e.reason)
+        raise
+    except urllib.error.URLError as e:
+        LOG.exception("URLError ({%s}): {%s}", req.full_url, e.reason)
+        raise
+
+
+def get_latest_channels(
+    num_of_channels: int, flavor: str, include_latest=True
+) -> List[str]:
```

Comment:
this needs architecture awareness:

```suggestion
def get_latest_channels(
    num_of_channels: int, flavor: str, arch: str, include_latest=True
) -> List[str]:
```